src/ipc/ipc-helper.c

```diff
@@ -174,8 +175,8 @@ int comp_buffer_connect(struct comp_dev *comp, uint32_t comp_core,
 {
 	/* check if it's a connection between cores */
 	if (buffer->core != comp_core) {
-		/* set the buffer as a coherent object */
-		coherent_shared_thread(buffer, c);
+		/* buffer must be shared */
+		assert(buffer->is_shared);
```

Comment:
Is this recoverable ? I assume this data is derived from topology so could a bad topology cause an assert here ?

Reply:
Not recoverable, must not happen. 
If happens, a cached alias will be used for shared memory and system will crash later

Reply:
but here at least you have a perfect chance to error out, don't you?